internal/oscap/oscap.go

```diff
@@ -78,88 +76,87 @@ func GetProfileDescription(profile string, datastream string) string {
 	return description[0] // get rid of new line
 }
 
-func BlueprintToCustomizations(profile string, description string, bp Blueprint) (v1.Customizations, error) {
-	// Convert the custom data structure into a `Customizations` object.
-	// This will be easier to handle in IB's API later on
-	customizations := v1.Customizations{}
-
-	var fs []v1.Filesystem
-	for _, bpFileSystem := range bp.Customizations.Filesystem {
-		fs = append(fs, v1.Filesystem{
-			MinSize:    bpFileSystem.Size,
-			Mountpoint: bpFileSystem.Mountpoint,
-		})
-	}
-	if len(fs) > 0 {
-		customizations.Filesystem = &fs
+func GenXMLTailoringFile(tailoring *string, datastream string) (*os.File, error) {
+	if tailoring == nil || *tailoring == "" {
+		// we don't need to process this any further,
+		// since the xml file will just end up blank
+		// and would cause issues later down the line
+		return nil, nil
 	}
 
-	var packages []string
-	for _, bpPackage := range bp.Packages {
-		packages = append(packages, bpPackage.Name)
+	jsonFile, err := os.CreateTemp("", "tailoring.json")
```

Comment:
`autotailor` doesn't seem to be in the `RHEL-9 ubi` so maybe we could just pass the `xml` tailoring in directly for this. Otherwise we could maybe pull in the binary from `fedora` in the `Dockerfile`, feels a bit gross though

Reply:
CreateTemp does open file for rw and returns a handle, the handle is not closed and then file is deleted which only means it is not available from the tree but since it is opened it looks like a file handle leak to me.